gobblin-core/src/main/java/org/apache/gobblin/source/extractor/extract/restapi/RestApiConnector.java

```diff
@@ -144,8 +142,8 @@ protected HttpClient getHttpClient() {
           .setStatePropertiesPrefix(ConfigurationKeys.SOURCE_CONN_PREFIX)
           .configure(this.state)
           .createClient();
-      if (httpClient instanceof CloseableHttpClient) {
-        this.closer.register((CloseableHttpClient)httpClient);
+      if (httpClient instanceof Closeable) {
```

Comment:
since the Closer is our only resource management angle, should we log when not a Closeable, to give some trace before falling through the cracks? better still would be to wrap within a holder that implements Closeable.

I did observe we still use the non-Closeable, AutoRetryHttpClient in some circumstances. if that's what we happen to have would the wrapper use its ClientConnectionManger to manage resources?